From 2660526b18cc9e2294e3f935c7c289ef56c76183 Mon Sep 17 00:00:00 2001
From: Herrie <Github.com@herrie.org>
Date: Mon, 2 Oct 2023 10:29:27 +0200
Subject: [PATCH] Update webos-connman-adapter.role.json.in: Add permissions to
 com.webos.service.downloadmanager|location and com.webos.lunasend-*

Fixes:

Oct 02 07:51:15 qemux86-64 ls-hubd[291]: [] [pmlog] ls-hubd MSG_TRUNCATED {"MSGID":"LSHUB_NO_NAME_PERMS","CAUSE":"Log message exceeded 1024 bytes","TRUNCATED_MSG":"Can not find match for 'com.webos.service.downloadmanager' in pattern queue '[\"org.webosports.wifi\", \"org.webosports.systemserv ..."}
Oct 02 07:51:15 qemux86-64 ls-hubd[291]: [] [pmlog] ls-hubd LSHUB_NO_NAME_PERMS {} Can not find match for 'com.webos.service.downloadmanager' in pattern queue '["org.webosports.wifi", "org.webosports.systemservice", "org.webosports.settingsservice", "org.webosports.service.wifi", "org.webosports.service.systemservice", "org.webosports.service.settingsservice", "org.webosports.service.pdm", "org.webosports.service.eventmonitor", "org.webosports.service.connectionmanager", "org.webosports.service.config", "org.webosports.service.activitymanager.client", "org.webosports.pdm", "org.webosports.eventmonitor", "org.webosports.connectionmanager", "org.webosports.config", "org.webosports.activitymanager.client", "org.webosinternals.wifi", "org.webosinternals.systemservice", "org.webosinternals.settingsservice", "org.webosinternals.service.wifi", "org.webosinternals.service.systemservice", "org.webosinternals.service.settingsservice", "org.webosinternals.service.pdm", "org.webosinternals.service.eventmonitor", "org.webosinternals.service.connectionmanager", "org.webosinternals.service.config", "o
Oct 02 07:51:15 qemux86-64 ls-hubd[291]: [] [pmlog] ls-hubd LSHUB_NO_OUT_PERMS {"DEST_APP_ID":"com.webos.service.downloadmanager","SRC_APP_ID":"com.webos.service.connectionmanager","EXE":"/usr/sbin/webos-connman-adapter","PID":522} "com.webos.service.connectionmanager" does not have sufficient outbound permissions to communicate with "com.webos.service.downloadmanager" (cmdline: /usr/sbin/webos-connman-adapter)

 ls-hubd LSHUB_NO_NAME_PERMS {} Can not find match for
 'com.webos.lunasend-1012' in pattern queue '["org.webosports.wifi",
 "org.webosports.systemservice", "org.webosports.settingsservice",
 "org.webosports.service.wifi", "org.webosports.service.systemservice",
 "org.webosports.service.settingsservice", "org.webosports.service.pdm",
 "org.webosports.service.eventmonitor",
 "org.webosports.service.downloadmanager",
 "org.webosports.service.connectionmanager",
 "org.webosports.service.config",
 "org.webosports.service.activitymanager.client", "org.webosports.pdm",
 "org.webosports.eventmonitor", "org.webosports.downloadmanager",
 "org.webosports.connectionmanager", "org.webosports.config",
 "org.webosports.activitymanager.client", "org.webosinternals.wifi",
 "org.webosinternals.systemservice",
 "org.webosinternals.settingsservice",
 "org.webosinternals.service.wifi",
 "org.webosinternals.service.systemservice",
 "org.webosinternals.service.settingsservice",
 "org.webosinternals.service.pdm",
 "org.webosinternals.service.eventmonitor", "org.webosinternals.s
 Oct 09 12:35:56 qemux86-64 ls-hubd[240]: [] [pmlog] ls-hubd
 LSHUB_NO_OUT_PERMS
 {"DEST_APP_ID":"com.webos.lunasend-1012","SRC_APP_ID":"com.webos.service.connectionmanager","EXE":"/usr/sbin/webos-connman-adapter","PID":411}
 "com.webos.service.connectionmanager" does not have sufficient outbound
 permissions to communicate with "com.webos.lunasend-1012" (cmdline:
 /usr/sbin/webos-connman-adapter)

Upstream-Status: Submitted [https://github.com/webosose/webos-connman-adapter/pull/1]

Signed-off-by: Herman van Hazendonk <github.com@herrie.org>
---
 files/sysbus/webos-connman-adapter.role.json.in | 15 ++++++++++++---
 1 file changed, 12 insertions(+), 3 deletions(-)

diff --git a/files/sysbus/webos-connman-adapter.role.json.in b/files/sysbus/webos-connman-adapter.role.json.in
index 917c391..9a54aac 100644
--- a/files/sysbus/webos-connman-adapter.role.json.in
+++ b/files/sysbus/webos-connman-adapter.role.json.in
@@ -10,10 +10,13 @@
                         "com.webos.service.config",
                         "com.webos.service.connectionmanager",
                         "com.webos.service.eventmonitor",
+                        "com.webos.service.location",
                         "com.webos.service.pdm",
                         "com.webos.service.systemservice",
                         "com.webos.service.wifi",
-                        "com.webos.settingsservice"]
+                        "com.webos.settingsservice",
+                        "com.webos.service.downloadmanager",
+                        "com.webos.lunasend-*"]
         },
         {
             "service":"com.webos.service.connectionmanager",
@@ -21,10 +24,13 @@
                         "com.webos.service.config",
                         "com.webos.service.connectionmanager",
                         "com.webos.service.eventmonitor",
+                        "com.webos.service.location",
                         "com.webos.service.pdm",
                         "com.webos.service.systemservice",
                         "com.webos.service.wifi",
-                        "com.webos.settingsservice"]
+                        "com.webos.settingsservice",
+                        "com.webos.service.downloadmanager",
+                        "com.webos.lunasend-*"]
         },
         {
             "service":"com.webos.service.wan",
@@ -32,10 +38,13 @@
                         "com.webos.service.config",
                         "com.webos.service.connectionmanager",
                         "com.webos.service.eventmonitor",
+                        "com.webos.service.location",
                         "com.webos.service.pdm",
                         "com.webos.service.systemservice",
                         "com.webos.service.wifi",
-                        "com.webos.settingsservice"]
+                        "com.webos.settingsservice",
+                        "com.webos.service.downloadmanager",
+                        "com.webos.lunasend-*"]
         }
     ]
 }
